connaisseur/flask_server.py

```diff
@@ -59,12 +60,61 @@ def mutate():
     return jsonify(response)
 
 
+# health probe
 @APP.route("/health", methods=["GET", "POST"])
 def healthz():
     """
-    Handles the '/health' path and checks the health status of the flask and
-    the notary server. Sends back either '200' for a healthy status or '500'
+    Handles the '/health' endpoint and checks the health status of the flask
+    server. Sends back either '200' for a healthy status or '500'
     otherwise.
     """
-    health = health_check(os.environ.get("NOTARY_SERVER"))
-    return ("", 200) if health else ("", 500)
+    return ("", 200)
+
+
+# readiness probe
+@APP.route("/ready", methods=["GET", "POST"])
+def readyz():
+    """
+    Handles the '/ready' endpoint and checks the health status of the configured notary
+    server and whether the webhook is installed or not. If the notary is available and the
+    webhook is installed it returns a 200 status code. Otherwise should one of them not be
+    reachable, 500 is returned. For installation purposes, the readiness probe first
+    checks whether a specific bootstrap pod is running in the namespace. If this pod can
+    be found and is still running, the readiness probe automatically returns 200. This
+    bootstrap pod will only run for the first 30 seconds after installation or until the
+    webhook is installed, after which the pod gets immediately deleted. From there on the
+    readiness probe checks the notary server and webhook as usual.
+    """
+    sentinel = os.environ.get("CONNAISSEUR_SENTINEL")
+    sentinel_ns = os.environ.get("CONNAISSEUR_NAMESPACE")
+    sentinel_path = "api/v1/namespaces/{ns}/pods/{name}".format(
+        ns=sentinel_ns, name=sentinel
+    )
+
+    sentinel_response = {}
+    try:
+        sentinel_response = api.request_kube_api(sentinel_path)
+    except Exception:  # nosec
```

Comment:
Is this supposed to catch the exception that's raised when the sentinel is not running? If so, we should be explicit about it. If not, what other exceptions do you have in mind?